### PR TITLE
Authenticate web sdk migration

### PR DESCRIPTION
**Pull request by ht-abuabraham**

### CODE REVIEW

File: src/services/authenticate.client.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+            mobileNumber: mobileNumber,
+        };
+
+        return this.postNoAuth(url, requestBody);
+    }
+
+    private setCurrentUser(username: string) {
+        if(username && this.currentUser !== username){
+            this.currentUser = username;
+            if ( !this.haventecAuthenticate ) {
```
[the remark this answers is not in the document]
Comment (justincrosbiehaventec) [reply src/services/authenticate.client.ts]: haventecAuthenticate is injected in via Angular instantiation which invokes the provider factory. So this logic could be removed. But its extra safety, so its ok to leave it in

File: src/services/http.service.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+@Injectable()
+export class HT_HttpService {
+
+    constructor(private http: Http) {
+    }
+
+    public get(url: string, access_token: string) {
+
+        var self = this;
+        return new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (justincrosbiehaventec) [reply src/services/http.service.ts]: See https://www.pivotaltracker.com/story/show/166404081

File: src/services/authenticate.client.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+            deviceUuid: this.haventecAuthenticate.getDeviceUuid(),
+            hashedPin: this.haventecAuthenticate.hashPin(pin),
+            username: username
+        };
+
+        return this.postNoAuth(url, requestBody);
+    }
+
+    public addDevice(username: string, deviceName?: string, urlOverwrite?:string) {
+        let url = (urlOverwrite ? urlOverwrite : this.domain + '/device');
```
[the remark this answers is not in the document]
Comment (justincrosbiehaventec) [reply src/services/authenticate.client.ts]: OOS, see https://www.pivotaltracker.com/story/show/166404081

File: src/services/authenticate.client.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+            deviceName = this.haventecAuthenticate.getDeviceName();
+        }
+
+        let requestBody = {
+            applicationUuid: this.applicationUuid,
+            deviceName: deviceName,
+            username: username
+        };
+
+        return this.postNoAuth(url, requestBody);
```
[the remark this answers is not in the document]
Comment (justincrosbiehaventec) [reply src/services/authenticate.client.ts]: OOS, see https://www.pivotaltracker.com/story/show/166404081

File: src/services/authenticate.client.ts [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+    constructor(private http:  HT_HttpService
+    ) {
+        
+    }
+
+    public init(applicationUuid: string, domain: string) {
+        this.applicationUuid = applicationUuid;
+        this.domain = domain;
+        let currentUser = localStorage.getItem('haventec_username');
```
[the remark this answers is not in the document]
Comment (justincrosbiehaventec) [reply src/services/authenticate.client.ts]: Actioned

File: src/services/authenticate.client.ts [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+            );
+        });
+    }
+
+    public getHashPin(pin: string): string {
+        return this.haventecAuthenticate.hashPin(pin);;
+    }
+
+    public getUsername() {
+        return this.haventecAuthenticate ? this.haventecAuthenticate.getUsername(): undefined;
```
[the remark this answers is not in the document]
Comment (justincrosbiehaventec) [reply src/services/authenticate.client.ts]: Actioned

**Review by htclifford — APPROVED** [review]

**Pull request merged**